piratebartender/js/app.js

```diff
@@ -0,0 +1,160 @@
+$(function() {
+
+
+    //ingredients constructor
+    var Ingredients = function(options) {
```

Comment:
By doing the following code you eliminate the need for the preceding code `var sweet = new .......` because you are now declaring the preference as part of the new object, not naming a variable the preference and storing a created object in it. So you could do the following now:

```
var ingredients = [
      new Ingredient('bunny ears', 'sweet');
      new Ingredient('muddled peacock feathers', 'sweet');
      new Ingredient('whipped cream', 'sweet');
      and so on and so forth....
]
```
`ingredients[]` now stores all your possible ingredients with the preferences tied to them. You can pass `ingredients[]` through the `Pantry` constructor. Then then when you create `var newPantry` it is an instance of the `Pantry` Constructor with all the `ingredients[]` present for later use.
You can do the same `this.preference` with the `Questions` Constructor.